build-system/tasks/check-links.js

```diff
@@ -16,88 +16,98 @@
 'use strict';
 
 const argv = require('minimist')(process.argv.slice(2));
-const BBPromise = require('bluebird');
 const fs = require('fs-extra');
 const log = require('fancy-log');
-const markdownLinkCheck = BBPromise.promisify(require('markdown-link-check'));
+const markdownLinkCheck = require('markdown-link-check');
 const path = require('path');
-const {
-  gitDiffAddedNameOnlyMaster,
-  gitDiffNameOnlyMaster,
-} = require('../common/git');
-const {green, magenta, red, yellow} = require('ansi-colors');
+const {getFilesToCheck} = require('../common/utils');
+const {gitDiffAddedNameOnlyMaster} = require('../common/git');
+const {green, cyan, red, yellow} = require('ansi-colors');
 const {isTravisBuild} = require('../common/travis');
+const {linkCheckGlobs} = require('../test-configs/config');
 const {maybeUpdatePackages} = require('./update-packages');
 
+let filesIntroducedByPr;
+
 /**
- * Parses the list of files in argv, or extracts it from the commit log.
- *
- * @return {!Array<string>}
+ * Checks for dead links in .md files passed in via --files or --local_changes.
  */
```

Comment:
Can you indicate the return type here? `Promise<void>`

Reply:
We normally omit the `@return` annotation for implicit promises returned by `async` functions. 

If there's good reason not to, I'm happy to send out a separate PR that makes our JSDoc checker enforce them (and auto-fix all missing instances). See `forceReturnsWithAsync` [here](https://www.npmjs.com/package/eslint-plugin-jsdoc#require-returns).